test/e2e/service.go

```diff
@@ -789,6 +789,65 @@ var _ = framework.KubeDescribe("Services", func() {
 		}
 	})
 
+	It("should be able to change the type from ExternalName to ClusterIP", func() {
+		serviceName := "externalname-service"
+		ns := f.Namespace.Name
+		jig := framework.NewServiceTestJig(cs, serviceName)
+
+		By("creating a service " + serviceName + " with the type=ExternalName in namespace " + ns)
+		externalNameService := jig.CreateExternalNameServiceOrFail(ns, nil)
+		jig.SanityCheckService(externalNameService, v1.ServiceTypeExternalName)
+		By("changing the ExternalName service to type=ClusterIP")
+		clusterIPService := jig.UpdateServiceOrFail(ns, externalNameService.Name, func(s *v1.Service) {
+			s.Spec.Type = v1.ServiceTypeClusterIP
+			s.Spec.ExternalName = ""
+			s.Spec.Ports = []v1.ServicePort{
+				{Port: 80, Name: "http", Protocol: "TCP"},
+			}
+		})
+		jig.SanityCheckService(clusterIPService, v1.ServiceTypeClusterIP)
+	})
+
+	It("should be able to change the type from ExternalName to NodePort", func() {
+		serviceName := "externalname-service"
+		ns := f.Namespace.Name
+		jig := framework.NewServiceTestJig(cs, serviceName)
+
+		By("creating a service " + serviceName + " with the type=ExternalName in namespace " + ns)
+		externalNameService := jig.CreateExternalNameServiceOrFail(ns, nil)
+		jig.SanityCheckService(externalNameService, v1.ServiceTypeExternalName)
+		By("changing the ExternalName service to type=NodePort")
+		nodePortService := jig.UpdateServiceOrFail(ns, externalNameService.Name, func(s *v1.Service) {
+			s.Spec.Type = v1.ServiceTypeNodePort
+			s.Spec.ExternalName = ""
+			s.Spec.Ports = []v1.ServicePort{
+				{Port: 80, Name: "http", Protocol: "TCP"},
+			}
+		})
+		jig.SanityCheckService(nodePortService, v1.ServiceTypeNodePort)
+	})
+
+	It("should be able to change the type from ExternalName to LoadBalancer", func() {
```

Comment:
Change this one to `ClusterIP to ExternalName. `

Creating LoadBalancer is slow.